test/test_ops.py

```diff
@@ -1107,114 +1107,149 @@ def test_bbox_convert_jit(self):
         torch.testing.assert_close(scripted_cxcywh, box_cxcywh, rtol=0.0, atol=TOLERANCE)
 
 
-class TestBoxArea:
-    def test_box_area(self):
-        def area_check(box, expected, tolerance=1e-4):
-            out = ops.box_area(box)
-            torch.testing.assert_close(out, expected, rtol=0.0, check_dtype=False, atol=tolerance)
-
-        # Check for int boxes
-        for dtype in [torch.int8, torch.int16, torch.int32, torch.int64]:
-            box_tensor = torch.tensor([[0, 0, 100, 100], [0, 0, 0, 0]], dtype=dtype)
-            expected = torch.tensor([10000, 0])
-            area_check(box_tensor, expected)
-
-        # Check for float32 and float64 boxes
-        for dtype in [torch.float32, torch.float64]:
-            box_tensor = torch.tensor(
-                [
-                    [285.3538, 185.5758, 1193.5110, 851.4551],
-                    [285.1472, 188.7374, 1192.4984, 851.0669],
-                    [279.2440, 197.9812, 1189.4746, 849.2019],
-                ],
-                dtype=dtype,
-            )
-            expected = torch.tensor([604723.0806, 600965.4666, 592761.0085], dtype=torch.float64)
-            area_check(box_tensor, expected, tolerance=0.05)
-
-        # Check for float16 box
-        box_tensor = torch.tensor(
-            [[285.25, 185.625, 1194.0, 851.5], [285.25, 188.75, 1192.0, 851.0], [279.25, 198.0, 1189.0, 849.0]],
-            dtype=torch.float16,
-        )
-        expected = torch.tensor([605113.875, 600495.1875, 592247.25])
-        area_check(box_tensor, expected)
-
-    def test_box_area_jit(self):
-        box_tensor = torch.tensor([[0, 0, 100, 100], [0, 0, 0, 0]], dtype=torch.float)
-        TOLERANCE = 1e-3
-        expected = ops.box_area(box_tensor)
-        scripted_fn = torch.jit.script(ops.box_area)
-        scripted_area = scripted_fn(box_tensor)
-        torch.testing.assert_close(scripted_area, expected, rtol=0.0, atol=TOLERANCE)
+class BoxTestBase(ABC):
+    @abstractmethod
+    def _target_fn(self) -> Tuple[bool, Callable]:
+        pass
 
+    def __perform_box_operation(self, box: Tensor, run_as_script: bool = False) -> Tensor:
```

Comment:
To avoid name mangling, I would recommend changing these to `_perform_box_operation`. Same for the rest of the methods that start with double underscore as we don't use this idiom in the codebase.